File: Solutions/121-BestTimeToBuyAndSellStock/BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;

        int leftPointer = 0;
        int rightPointer = 1;

        while (rightPointer < prices.length) {
            if (prices[leftPointer] < prices[rightPointer]) {
                int profit = prices[rightPointer] - prices[leftPointer];
                if (profit > maxProfit) {
                    maxProfit = profit;
                }
            } else {
                leftPointer = rightPointer;
            }
            rightPointer++;
        }

        return maxProfit;
    }
}

File: Solutions/121-BestTimeToBuyAndSellStock/BestTimeToBuyAndSellStock.ts
function maxProfit(prices: number[]): number {
	let p1 = 0;
	let p2 = 0;
	let maxProfit = 0;

	while (p2 < prices.length) {
		if (prices[p1] < prices[p2]) {
			maxProfit = Math.max(maxProfit, prices[p2] - prices[p1]);
		} else p1 = p2;

		p2++;
	}

	return maxProfit;
}
